pallets/utility/src/lib.rs

```diff
@@ -156,5 +184,55 @@ decl_module! {
 
             Ok(())
         }
+
+        /// Relay a call for a target from an origin
+        ///
+        /// Relaying in this context refers to the ability of origin to make a call on behalf of
+        /// target.
+        ///
+        /// Fees are charged to origin
+        ///
+        /// # Parameters
+        /// - `target`: Account to be relayed
+        /// - `signature`: Signature from target authorizing the relay
+        /// - `call`: Call to be relayed on behalf of target
+        ///
+        /// # Weight
+        /// - The weight of the call to be relayed plus a static 250_000.
+        #[weight = (
+            call.call.get_dispatch_info().weight.saturating_add(250_000),
+            call.call.get_dispatch_info().class,
+        )]
+        pub fn relay_tx(
+            origin,
+            target: T::AccountId,
+            signature: <T as IdentityTrait>::OffChainSignature,
+            call: UniqueCall<<T as Trait>::Call>
+        ) -> DispatchResult {
+            let _ = ensure_signed(origin)?;
+
+           let target_nonce = <Nonces<T>>::get(&target);
+
+            ensure!(
+                target_nonce == call.nonce,
+                Error::<T>::InvalidNonce
+            );
+
+            ensure!(
+                signature.verify(call.encode().as_slice(), &target),
+                Error::<T>::InvalidSignature
+            );
+
+            ensure!(
+                T::CddChecker::check_key_cdd(&target),
+                Error::<T>::TargetCddMissing
+            );
+
+            <Nonces<T>>::insert(target.clone(), target_nonce + 1);
+
+            call.call.dispatch(RawOrigin::Signed(target).into())
```

Comment:
I completely forgot to mention that if we don't handle the post_dispatch_info here in a right manner then gas refunds will not work. Right now we don't have extensive features for the weight so we can ignore it for now otherwise it is very important to handle.

@CJP10 could you please add a TODO over the dispatchable

Reply:
or it may be good to apply those changes in this PR as well. You can take a look here for the reference https://github.com/PolymathNetwork/Polymesh/blob/fdf6f5a15d1b575e3fb339e03d8be056c53f2fe5/pallets/identity/src/lib.rs#L555